deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,138 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing a check for computing a scalar performance metric for a single dataset."""
+import typing as t
+import warnings
+
+import torch
+from ignite.metrics import Accuracy, Metric
+
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+from deepchecks.vision.vision_data import TaskType
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric, default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+    """
+
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+
+        self.metric_name = kwargs.get('metric_name') or (metric.__class__.__name__ if metric else None)
+        self.reduce_name = kwargs.get('reduce_name') or (reduce.__name__ if reduce else None)
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+                if self.metric_name is None:
+                    self.metric_name = 'accuracy'
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                if self.metric_name is None:
```

Comment:
I think user shouldn't be able to pass a name if he didn't pass a metric.

Reply:
+1 also, why even pass a metric_name? can't we just use the class name and maybe just split by camel case if we really want

Reply:
I think the user should be able to pass a name because they might want to use the default metric with a different name like "default" or something else according to their conventions.